doc/api/http2.md

```diff
@@ -0,0 +1,727 @@
+# HTTP2
+
+> Stability: 1 - Experimental
+
+The `http2` module provides an implementation of the [HTTP/2][] protocol. It
+can be accessed using:
+
+```js
+const http2 = require('http2');
+```
```

Comment:
if we are not shipping this in `8.0.0`, we need to ship it behind a flag, as we will be shadowing the `'http2'` module otherwise. Maybe we might do something different here.